### ISSUE TITLE
Feature request: Keep dates

### ISSUE THREAD

**Issue by heinowalther** [sahib/rmlint#366]
Hello there

I have been using rmlint for a few years now, and I think it is great!
One little thing that I would like is if it somehow was able to avoid the date change of the folder it has deleted files from... I know this is a part of the filesystem and it is how it is supposed to work, but I would very much like to keep the dates on my folders... 
I think it should be fairly easy to implement, but I am aware that it will most likely not get the highest priority ;-)

Thanks for a great tool!

/Heino

**Comment by sahib** [comment]
> I have been using rmlint for a few years now, and I think it is great!

I just realized it has been roughly 9 years since ``rmlint`` was born. Time flies.

> One little thing that I would like is if it somehow was able to avoid the date change of the folder it has deleted files from... 

What would be your specific use case on that?

I think it should be fairly easy to implement, but I am aware that it will most likely not get the highest priority ;-)

> I think it should be fairly easy to implement [...]

I'm not quite sure about that. We do something similar already for normal files, but directories are harder.

**Comment by heinowalther** [comment]
The use case is that we have directories where we sort by date, to find the latest one...
We then sometimes add older folders into the structure (keeping the old date on the folder)
We then run the rmlint to remove any duplicates, and if it finds one, it will delete it, and the folder will show up as a new folder because of it...

If you could just record the directory date where a file is located before a file it is deleted, and the alter the directory date, it would solve the issue...  but again I'm not sure how easy it is to implement, yet if you are able to do so great this far, making this little thing should be no problem? ;-)

As I mentioned before, I understand if it gets low priority :-)

/Heino

**Comment by SeeSpotRun** [comment]
It would be relatively easy to do by modifying the shell script, similar to https://unix.stackexchange.com/questions/445419/how-to-remove-temp-files-without-changing-the-directorys-modification-timestamp

Even without modifying `rmlint` you could do something like:
```
$ rmlint -o sh -c sh:cmd='echo "Silently deleting $1" && rm-silent "$1"'
```

... and create a shell script in your path something like:
```
rm-silent:
---------
#!/bin/sh
STAMPFILE=/tmp/rmlint-timestamp
DIR=$(dirname "$1")
touch -r "$DIR" "$STAMPFILE"
rm "$1"
touch -r "$STAMPFILE" "$DIR"
rm "$STAMPFILE"
```

For a more permanent change, modify `remove_cmd()` in https://github.com/sahib/rmlint/blob/master/lib/formats/sh.sh to use rm-silent

**Comment by sahib** [comment]
> It would be relatively easy to do by modifying the shell script

Okay, right. I was a bit worried about parallel changes to directories, but it seems I was overthinking.
I implemented this approach in 41384e82eabace2ec67d2beeee1c0647d6c93daf. @heinowalther, please try it out and see if works for you.

**Comment by sahib** [comment]
I will close this issue in a few days, if there are no new answers.
